server/segment_rpc.go

```diff
@@ -340,8 +341,21 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 		return nil, err
 	}
 
+	// set a minimum timeout to accommodate transport / processing overhead
+	dur := common.HTTPTimeout
+	paddedDur := 4.0 * seg.Duration // use a multiplier of 4 for now
+	if paddedDur > dur.Seconds() {
+		dur, err = time.ParseDuration(fmt.Sprintf("%fs", paddedDur))
```

Comment:
Why ParseDuration here? Why not
`dur = time.Duration(paddedDur * float64(time.Second))` ?

Reply:
Good question. I actually remember spending some time on this particular line, but I can't for the life of me remember the reason for that. It might have to do with invalid value handling, but again don't remember the specific case (float range? seems unlikely / irrelevant).

Anyway, I agree this approach is better, and it doesn't seem to break anything else, so corrected in 856ceaf836dc780c454d476acd68b94c7cb2aab7